app/javascript/src/ActiveDocs/components/ActiveDocsSpec.jsx

```diff
@@ -4,15 +4,84 @@ import React from 'react'
 
 import { createReactWrapper } from 'utilities/createReactWrapper'
 import SwaggerUI from 'swagger-ui-react'
-
 import 'swagger-ui-react/swagger-ui.css'
 
 type ActiveDocsSpecProps = {
-  url: string
+  accountData: {
+    [string]: Array<mixed>
+  },
+  url: string,
+}
+
+const AUTOCOMPLETE_PARAMS_MAP = {
+  'app_key': 'app_keys',
+  'app_id': 'app_ids',
+  'application_id': 'application_ids',
+  'user_key': 'user_keys',
+  'user_id': 'user_ids',
+  'account_id': 'account_ids',
+  'metric_name': 'metric_names',
+  'metric_id': 'metric_ids',
+  'backend_api_metric_name': 'backend_api_metric_names',
+  'service_id': 'service_ids',
+  'admin_id': 'admin_ids',
+  'service_plan_id': 'service_plan_ids',
+  'application_plan_id': 'application_plan_ids',
+  'account_plan_id': 'account_plan_ids',
+  'client_id': 'client_ids',
+  'client_secret': 'client_secrets',
+  'service_token': 'service_tokens',
+  'access_token': 'access_token'
+}
+
+const getParam = (param, accountData) => {
+  const paramName = AUTOCOMPLETE_PARAMS_MAP[param.name]
+  const paramHasAutocompleteData = (accountData[paramName].length > 0) &&
+    accountData[paramName].every(param => param.value !== '')
+  if (!paramHasAutocompleteData) {
+    return param
+  }
+  const autocompleteData = accountData[paramName]
+  const examples = { '0': { summary: 'Select an option to autocomplete', value: '-' } }
+  autocompleteData.forEach((item, index) => {
+    examples[`${index + 1}`] = {
+      summary: item.name,
+      value: item.value
+    }
+  })
+  const paramWithExamples = {
+    ...param,
+    examples
+  }
+  return paramWithExamples
 }
 
-const ActiveDocsSpec = ({ url }: ActiveDocsSpecProps) => <SwaggerUI url={url} />
+const injectAccountDataToResponse = (response, accountData) => {
+  const responseWithAutocomplete = Object.assign({}, response)
+  const paths = Object.assign({}, response.body.paths)
+  let pathsWithAutocomplete = {}
+
+  for (let key in paths) {
+    pathsWithAutocomplete[key] = paths[key]
+    const parametersWithAutocomplete = paths[key].parameters
+      .map(param => getParam(param, accountData))
+    pathsWithAutocomplete[key].parameters = parametersWithAutocomplete
+  }
+  responseWithAutocomplete.body.paths = pathsWithAutocomplete
+  responseWithAutocomplete.data = JSON.stringify(responseWithAutocomplete.body)
+  responseWithAutocomplete.text = JSON.stringify(responseWithAutocomplete.body)
+  return responseWithAutocomplete
```

Comment:
So the entire idea is to replace the response with the autocomplete values. What's the type of this? I'm sure there's a way to simplify the logic of crafting the response with autocomplete